feat: add cli arguments

noomzopendream opened the pull request:
- close #8

@mattparkerls plz check if its run great and correct.

noomzopendream committed:
feat: add cli config

noomzopendream committed:
feat: remove hard-code dG comparing

noomzopendream committed:
feat: use more subtle way to find vendor binary

noomzopendream committed:
fix: clappy and fmt

noomzopendream committed:
feat: allow pass cli args with ENV

mattparkerls commented on od-msspe/src/config.rs:
```diff
@@ -0,0 +1,177 @@
+        value_parser = ["yes", "no"],
+        help = "Removes primers with melting temperature greater than 2 standard deviations from mean of the Tm values for all primers in set. Default is true."
+    )]
+    pub strict_tm_range: String,
```
@noomzopendream Needs a line break

mattparkerls commented on od-msspe/src/config.rs:
```diff
@@ -0,0 +1,177 @@
+    #[arg(long, env = "DELTA_G_THRESHOLD", default_value_t = DELTA_G_THRESHOLD, help = "Threshold for dG, default is -9000.0 J/mol")]
+    pub delta_g_threshold: f32,
+
+    // logic based config
```
change all logic based config to true/false

yes/no is not best practice @noomzopendream

mattparkerls commented on od-msspe/src/main.rs:
```diff
@@ -299,114 +282,529 @@ fn find_most_freq_kmer<'a>(
+                || (kmer_stat.self_end_th < primer_config.max_self_dimer_end_tm);
+            let pass_hairpin = !program_config.check_hairpin
+                || (kmer_stat.hairpin_th < primer_config.max_hairpin_tm);
+
```
unnecessary line break

mattparkerls commented:
Overall great job and code looks good – a few small changes:

1. All booleans should be "true\false" not "yes/no"
2. if I set the cli argument "strict-tm-range=yes" then the other filter "max-tm" doesn't work/return any primers if I set it lower than the default (<60). If I set "strict-tm-range=false" then I can use the "max-tm" but then it doesn't calculate any tm stats. We want users to be able to just set min & max tms to filter all primers between a range of tms. Then separately, they can ignore the strict/standard deviation filtering if they want. 

@noomzopendream

noomzopendream commented:
@mattparkerls OK. I will look at it.

noomzopendream committed:
chore: wrap help text

noomzopendream committed:
chore: change yes/no args value to true/false

noomzopendream committed:
fix: `--max-tm` arg doesn't work.

Must provide more opt for primer3, see https://primer3.org/manual#PRIMER_OPT_TM

noomzopendream committed:
chore: remove unnecessary whitespace in config.rs

noomzopendream commented:
@mattparkerls Fixed! PRIMER_OPT_TM is a cause, see more here https://primer3.org/manual#PRIMER_OPT_TM

Commit:
made min or max tm filters independent of strict-tm-range filter

noomzopendream committed:
chore: fix fmt

noomzopendream committed:
feat: add tm_stddev and disable_tm_stddev options for primer configuration
